### ISSUE TITLE
Allow seeking within audio

### ISSUE THREAD

**Issue by jD91mZM2** [jhasse/ears#15]
Something like a `AudioControllable.seek` would be amazing. Any change to get it implemented? It'd allow making applications that can move back and forth in audio, like a music player. Cheers!

**Comment by jhasse** [comment]
Just noticed that someone started working on this: https://github.com/hhatinen/ears/commit/693fef58943e4f2d1858c821e51d4bd7f9487dad
@hhatinen Are you planning to also create a PR? :)

**Comment by hnen** [comment]
Yeah, I haven't tested the changes properly and currently am just playing
around. OpenAL seems to fail silently at random when seeking with alSourcef()
if the source is not exactly in the correct state. Also would be nice to
have the seek method on AudioController trait instead of Sound struct and
support it with Music as well (seeking is probably mostly used with music
anyway.)

But yeah if I continue using ears on my project I probably will put some
effort making the routines stable and implement it for Music as well since
I need seeking for music asset anyways. I can do PR even before that but
don't feel 100% comfortable putting in code that doesn't necessarily work
stable enough.

**Comment by jhasse** [comment]
Understandable. Thanks! (and sorry for pinging you out of the blue)

**Comment by hnen** [comment]
No problem, happy to collaborate :) Will let you know if/when I get to
finalize the changes.
